### PR TITLE
Imbue Network: milestone 1 review

### PR DESCRIPTION
**Pull request by samelamin**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/703

### CODE REVIEW

**Comment by samelamin** [comment]
FYI the PR for the milestone can be found here 

https://github.com/w3f/Grants-Program/pull/703

**Comment by takahser** [comment]
@samelamin thanks for the delivery. We will look into it as soon as possible.

**Comment by samelamin** [comment]
> @samelamin thanks for the delivery. We will look into it as soon as possible.

Thank you so much @takahser, I really am keen to hearing the feedback!

**Comment by takahser** [comment]
Hi @samelamin
Thanks for your patience. I took a first look at your delivery and noticed that the formatting on the _Grants Pallet_ delivery is off:
image

Also, unfortunately I couldn't find delivery 0e. | Article. May I ask you to address these two issues?

**Comment by samelamin** [comment]
Hi @takahser No worries, totally understand you folks might be busy. 

Fixed the formating issue

I know the article is missing but as I mentioned in the comments section, we felt it would make more sense to have an article once the UI is fully built out to include contributing/voting as well as milestone submission

Right now the only part of the UI that is done is the creation of the project

I hope that isn't a blocker for the milestone given what we have achieved from milestone 2. Really the main reason we applied for the delivery to help us get into the substrate builders programme and start talking about how we can build the next stage because we want to build the best thing possible for the ecosystem

> Hi @samelamin Thanks for your patience. I took a first look at your delivery and noticed that the formatting on the _Grants Pallet_ delivery is off: image
> 
> Also, unfortunately I couldn't find delivery _0e. | Article_. I'd assume you forgot to include it in which case I'd like to ask you to add it to the table and update the PR.

**Comment by takahser** [comment]
Hi @samelamin 
Thanks for elaborating on this.
Since you move milestone _0e. | Article_ from M1 to M2 we require you to do an [amendment](https://github.com/w3f/Grants-Program#changes-to-a-grant-after-approval) to your application. You don't have to change the costs since it's a minor change and I saw you already posted a couple of other articles on your Medium.

**Comment by samelamin** [comment]
Yes we have, but I also pushed a PR to move the article to Milestone 2



> Hi @samelamin Thanks for elaborating on this. Since you move milestone _0e. | Article_ from M1 to M2 we require you to do an [amendment](https://github.com/w3f/Grants-Program#changes-to-a-grant-after-approval) to your application. You don't have to change the costs since it's a minor change and I saw you already posted a couple of other articles on your Medium.

**Comment by takahser** [comment]
@samelamin thanks for the amendment, I'm going to have a deeper look at your delivery and get back to you soon.

**Comment by takahser** [comment]
Hey @samelamin

I'm a bit confused about when the funds for a certain milestone get unlocked.
In [L137](https://github.com/w3f/Grants-Program/blob/75e0aca0191c2ae309e10a59d95f386eca96e73f/applications/imbue_network.md?plain=1#L137) of your application it says that
> the council will determine when to mark a milestone as complete

Contrary to that, on the webpage and on [L29](https://github.com/w3f/Grants-Program/blob/75e0aca0191c2ae309e10a59d95f386eca96e73f/applications/imbue_network.md?plain=1#L29) it says that this is the role of the original funders:
> only the original funders can then vote to validate whether the milestone was completed successfully

Can you clearify who is supposed to unlock the funds upon milestone completion: The DAO Council or the Original Funders?

**Comment by samelamin** [comment]
> Hey @samelamin
> 
> I'm a bit confused about when the funds for a certain milestone get unlocked. In [L137](https://github.com/w3f/Grants-Program/blob/75e0aca0191c2ae309e10a59d95f386eca96e73f/applications/imbue_network.md?plain=1#L137) of your application it says that
> 
> > the council will determine when to mark a milestone as complete
> 
> Contrary to that, on the webpage and on [L29](https://github.com/w3f/Grants-Program/blob/75e0aca0191c2ae309e10a59d95f386eca96e73f/applications/imbue_network.md?plain=1#L29) it says that this is the role of the original funders:
> 
> > only the original funders can then vote to validate whether the milestone was completed successfully
> 
> Can you clearify who is supposed to unlock the funds upon milestone completion: The DAO Council or the Original Funders?

Hi @takahser 

Apologies for the confusion. Essentially we have two ways for a milestone to be approved

1) The initiator can [finalise](https://github.com/ImbueNetwork/imbue/blob/main/pallets/proposals/src/lib.rs#L641) the voting early assuming all the funders have voted 


2) Council can [approve](https://github.com/ImbueNetwork/imbue/blob/main/pallets/proposals/src/lib.rs#L448) a milestone


The reason we have the second is that we are still not clear when a milestone vote can be considered complete. i.e. only 70% or 80% of the voters have voted


But when it comes to withdrawing the approved funds, that's done by the project initator

**Comment by takahser** [comment]
Hey @samelamin 
Thanks for the super-quick response.
I see, so the council acts more like a backup instance in case, not all of the contributors vote (which is a fairly probable assumption to make I'd suppose). Just thinking out loud here, but would it maybe make sense to make the threshold for "% of contributor's votes required to complete a milestone" configurable? That way the initiators could withdraw without the approval of the council if enough (threshold reached) people have voted.
Note this won't be relevant for the acceptance of the milestone, it's just an idea. :)

**Comment by samelamin** [comment]
> Hey @samelamin Thanks for the super-quick response. I see, so the council acts more like a backup instance in case, not all of the contributors vote (which is a fairly probable assumption to make I'd suppose). Just thinking out loud here, but would it maybe make sense to make the threshold for "% of contributor's votes required to complete a milestone" configurable? That way the initiators could withdraw without the approval of the council if enough (threshold reached) people have voted. Note this won't be relevant for the acceptance of the milestone, it's just an idea. :)

Haha we are all keen to get this passed, so the full team is on hand to answer any questions or make any changes


Yeah of course the threshold can be made configurable, its probably best practice to have it that way 

We just need feedback on what that value should be. Once its set we can remove the need for the council in this decision entirely

**Comment by samelamin** [comment]
> > Hey @samelamin Thanks for the super-quick response. I see, so the council acts more like a backup instance in case, not all of the contributors vote (which is a fairly probable assumption to make I'd suppose). Just thinking out loud here, but would it maybe make sense to make the threshold for "% of contributor's votes required to complete a milestone" configurable? That way the initiators could withdraw without the approval of the council if enough (threshold reached) people have voted. Note this won't be relevant for the acceptance of the milestone, it's just an idea. :)
> 
> Haha we are all keen to get this passed, so the full team is on hand to answer any questions or make any changes
> 
> Yeah of course the threshold can be made configurable, its probably best practice to have it that way
> 
> We just need feedback on what that value should be. Once its set we can remove the need for the council in this decision entirely

**Comment by takahser** [comment]
Hey @samelamin 
When running the containers with `docker-compose up -d` in the dapp repo the build fails since `python` is missing within the api container:
```bash
% docker-compose up -d

WARNING: The GOOGLE_OAUTH2_CLIENT_ID variable is not set. Defaulting to a blank string.
WARNING: The GOOGLE_OAUTH2_CLIENT_SECRET variable is not set. Defaulting to a blank string.
Building web
[+] Building 24.4s (10/16)                                                                                                           
 => [internal] load build definition from Dockerfile                                                                            0.0s
 => => transferring dockerfile: 37B                                                                                             0.0s
 => [internal] load .dockerignore                                                                                               0.0s
 => => transferring context: 2B                                                                                                 0.0s
 => [internal] load metadata for docker.io/library/node:16-slim                                                                 1.6s
 => [internal] load metadata for docker.io/library/nginx:latest                                                                 1.6s
 => [internal] load build context                                                                                               0.0s
 => => transferring context: 36.61kB                                                                                            0.0s
 => [builder 1/5] FROM docker.io/library/node:16-slim@sha256:f527a6118422b888c35162e0a7e2fb2febced4c85a23d96e1342f9edc2789fec   0.0s
 => CACHED [stage-1 1/6] FROM docker.io/library/nginx:latest@sha256:2834dc507516af02784808c5f48b7cbe38b8ed5d0f4837f16e78d00deb  0.0s
 => CACHED [builder 2/5] WORKDIR /builds/imbue                                                                                  0.0s
 => CACHED [builder 3/5] ADD . .                                                                                                0.0s
 => ERROR [builder 4/5] RUN yarn install                                                                                       22.8s
------                                                                                                                               
 > [builder 4/5] RUN yarn install:                                                                                                   
#10 0.259 yarn install v1.22.15                                                                                                      
#10 0.309 [1/4] Resolving packages...                                                                                                
#10 0.416 [2/4] Fetching packages...                                                                                                 
#10 19.75 [3/4] Linking dependencies...                                                                                              
#10 19.76 warning " > @polkadot/extension-dapp@0.42.4" has unmet peer dependency "@polkadot/util@*".
#10 19.76 warning " > @polkadot/extension-dapp@0.42.4" has unmet peer dependency "@polkadot/util-crypto@*".
#10 22.29 [4/4] Building fresh packages...
#10 22.45 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
#10 22.45 error /builds/imbue/node_modules/utf-8-validate: Command failed.
#10 22.45 Exit code: 1
#10 22.45 Command: node-gyp-build
#10 22.45 Arguments: 
#10 22.45 Directory: /builds/imbue/node_modules/utf-8-validate
#10 22.45 Output:
#10 22.45 gyp info it worked if it ends with ok
#10 22.45 gyp info using node-gyp@8.3.0
#10 22.45 gyp info using node@16.13.2 | linux | arm64
#10 22.45 gyp ERR! find Python 
#10 22.45 gyp ERR! find Python Python is not set from command line or npm configuration
#10 22.45 gyp ERR! find Python Python is not set from environment variable PYTHON
#10 22.45 gyp ERR! find Python checking if "python3" can be used
#10 22.45 gyp ERR! find Python - "python3" is not in PATH or produced an error
#10 22.45 gyp ERR! find Python checking if "python" can be used
#10 22.45 gyp ERR! find Python - "python" is not in PATH or produced an error
#10 22.45 gyp ERR! find Python 
#10 22.45 gyp ERR! find Python **********************************************************
#10 22.45 gyp ERR! find Python You need to install the latest version of Python.
#10 22.45 gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
#10 22.45 gyp ERR! find Python you can try one of the following options:
#10 22.45 gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
#10 22.45 gyp ERR! find Python   (accepted by both node-gyp and npm)
#10 22.45 gyp ERR! find Python - Set the environment variable PYTHON
#10 22.45 gyp ERR! find Python - Set the npm configuration variable python:
#10 22.45 gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
#10 22.45 gyp ERR! find Python For more information consult the documentation at:
#10 22.45 gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
#10 22.45 gyp ERR! find Python **********************************************************
#10 22.45 gyp ERR! find Python 
#10 22.45 gyp ERR! configure error 
#10 22.45 gyp ERR! stack Error: Could not find any Python installation to use
#10 22.45 gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:330:47)
#10 22.45 gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:159:21)
#10 22.45 gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:202:16)
#10 22.45 gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:294:16)
#10 22.45 gyp ERR! stack     at exithandler (node:child_process:404:5)
#10 22.45 gyp ERR! stack     at ChildProcess.errorhandler (node:child_process:416:5)
#10 22.45 gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
#10 22.45 gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
#10 22.45 gyp ERR! stack     at onErrorNT (node:internal/child_process:477:16)
#10 22.45 gyp ERR! stack     at processTicksAndRejections (node:internal/process/task_queues:83:21)
#10 22.45 gyp ERR! System Linux 5.10.76-linuxkit
#10 22.45 gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
#10 22.45 gyp ERR! cwd /builds/imbue/node_modules/utf-8-validate
#10 22.45 gyp ERR! node -v v16.13.2
#10 22.45 gyp ERR! node-gyp -v v8.3.0
#10 22.45 gyp ERR! not ok
------
executor failed running [/bin/sh -c yarn install]: exit code: 1
ERROR: Service 'web' failed to build : Build failed
```

I already tried:
- installing it manually as part of the build process (Dockerfile) and 
- using the default `FROM node:16` image instead of the `-slim` one that [doesn't contain python](https://github.com/nodejs/docker-node/issues/12#issuecomment-84025397)

But unfortunately, neither of these worked. Are you able to run the containers on your machine with the current config? I'm currently on `44d319883e1ece693fd2bc96977453750abe4346`.

**Comment by samelamin** [comment]
> Hey @samelamin When running the containers with `docker-compose up -d` in the dapp repo the build fails since `python` is missing within the api container:
> 
> ```shell
> % docker-compose up -d
> 
> WARNING: The GOOGLE_OAUTH2_CLIENT_ID variable is not set. Defaulting to a blank string.
> WARNING: The GOOGLE_OAUTH2_CLIENT_SECRET variable is not set. Defaulting to a blank string.
> Building web
> [+] Building 24.4s (10/16)                                                                                                           
>  => [internal] load build definition from Dockerfile                                                                            0.0s
>  => => transferring dockerfile: 37B                                                                                             0.0s
>  => [internal] load .dockerignore                                                                                               0.0s
>  => => transferring context: 2B                                                                                                 0.0s
>  => [internal] load metadata for docker.io/library/node:16-slim                                                                 1.6s
>  => [internal] load metadata for docker.io/library/nginx:latest                                                                 1.6s
>  => [internal] load build context                                                                                               0.0s
>  => => transferring context: 36.61kB                                                                                            0.0s
>  => [builder 1/5] FROM docker.io/library/node:16-slim@sha256:f527a6118422b888c35162e0a7e2fb2febced4c85a23d96e1342f9edc2789fec   0.0s
>  => CACHED [stage-1 1/6] FROM docker.io/library/nginx:latest@sha256:2834dc507516af02784808c5f48b7cbe38b8ed5d0f4837f16e78d00deb  0.0s
>  => CACHED [builder 2/5] WORKDIR /builds/imbue                                                                                  0.0s
>  => CACHED [builder 3/5] ADD . .                                                                                                0.0s
>  => ERROR [builder 4/5] RUN yarn install                                                                                       22.8s
> ------                                                                                                                               
>  > [builder 4/5] RUN yarn install:                                                                                                   
> #10 0.259 yarn install v1.22.15                                                                                                      
> #10 0.309 [1/4] Resolving packages...                                                                                                
> #10 0.416 [2/4] Fetching packages...                                                                                                 
> #10 19.75 [3/4] Linking dependencies...                                                                                              
> #10 19.76 warning " > @polkadot/extension-dapp@0.42.4" has unmet peer dependency "@polkadot/util@*".
> #10 19.76 warning " > @polkadot/extension-dapp@0.42.4" has unmet peer dependency "@polkadot/util-crypto@*".
> #10 22.29 [4/4] Building fresh packages...
> #10 22.45 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
> #10 22.45 error /builds/imbue/node_modules/utf-8-validate: Command failed.
> #10 22.45 Exit code: 1
> #10 22.45 Command: node-gyp-build
> #10 22.45 Arguments: 
> #10 22.45 Directory: /builds/imbue/node_modules/utf-8-validate
> #10 22.45 Output:
> #10 22.45 gyp info it worked if it ends with ok
> #10 22.45 gyp info using node-gyp@8.3.0
> #10 22.45 gyp info using node@16.13.2 | linux | arm64
> #10 22.45 gyp ERR! find Python 
> #10 22.45 gyp ERR! find Python Python is not set from command line or npm configuration
> #10 22.45 gyp ERR! find Python Python is not set from environment variable PYTHON
> #10 22.45 gyp ERR! find Python checking if "python3" can be used
> #10 22.45 gyp ERR! find Python - "python3" is not in PATH or produced an error
> #10 22.45 gyp ERR! find Python checking if "python" can be used
> #10 22.45 gyp ERR! find Python - "python" is not in PATH or produced an error
> #10 22.45 gyp ERR! find Python 
> #10 22.45 gyp ERR! find Python **********************************************************
> #10 22.45 gyp ERR! find Python You need to install the latest version of Python.
> #10 22.45 gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
> #10 22.45 gyp ERR! find Python you can try one of the following options:
> #10 22.45 gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
> #10 22.45 gyp ERR! find Python   (accepted by both node-gyp and npm)
> #10 22.45 gyp ERR! find Python - Set the environment variable PYTHON
> #10 22.45 gyp ERR! find Python - Set the npm configuration variable python:
> #10 22.45 gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
> #10 22.45 gyp ERR! find Python For more information consult the documentation at:
> #10 22.45 gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
> #10 22.45 gyp ERR! find Python **********************************************************
> #10 22.45 gyp ERR! find Python 
> #10 22.45 gyp ERR! configure error 
> #10 22.45 gyp ERR! stack Error: Could not find any Python installation to use
> #10 22.45 gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:330:47)
> #10 22.45 gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:159:21)
> #10 22.45 gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:202:16)
> #10 22.45 gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:294:16)
> #10 22.45 gyp ERR! stack     at exithandler (node:child_process:404:5)
> #10 22.45 gyp ERR! stack     at ChildProcess.errorhandler (node:child_process:416:5)
> #10 22.45 gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
> #10 22.45 gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
> #10 22.45 gyp ERR! stack     at onErrorNT (node:internal/child_process:477:16)
> #10 22.45 gyp ERR! stack     at processTicksAndRejections (node:internal/process/task_queues:83:21)
> #10 22.45 gyp ERR! System Linux 5.10.76-linuxkit
> #10 22.45 gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
> #10 22.45 gyp ERR! cwd /builds/imbue/node_modules/utf-8-validate
> #10 22.45 gyp ERR! node -v v16.13.2
> #10 22.45 gyp ERR! node-gyp -v v8.3.0
> #10 22.45 gyp ERR! not ok
> ------
> executor failed running [/bin/sh -c yarn install]: exit code: 1
> ERROR: Service 'web' failed to build : Build failed
> ```
> 
> I already tried:
> 
> * installing it manually as part of the build process (Dockerfile) and
> * using the default `FROM node:16` image instead of the `-slim` one that [doesn't contain python](https://github.com/nodejs/docker-node/issues/12#issuecomment-84025397)
> 
> But unfortunately, neither of these worked. Are you able to run the containers on your machine with the current config? I'm currently on `44d319883e1ece693fd2bc96977453750abe4346`.

Hi @takahser 

Are you using an M1 Mac by any chance? 

I think we have seen this issue before, its part of the reason I want us to deploy an image for the dapp to a registry so we don't have to build it



That said, our CI/CD [pipeline](https://github.com/ImbueNetwork/dapp/blob/main/.github/workflows/dapp-deploy.yml#L47) does a clean checkout and rebuilds 

So perhaps try and reuse these [steps](https://github.com/ImbueNetwork/dapp/blob/main/scripts/deploy_to_aws.sh#L15)?

**Comment by takahser** [comment]
> Are you using an M1 Mac by any chance?

Yes, indeed.
I confirm that I was able to start the containers on a legacy (non-M1) macbook instead:
```bash
$ docker ps
CONTAINER ID   IMAGE                COMMAND                  CREATED          STATUS          PORTS                                          NAMES
851efd178a22   dapp_web             "/docker-entrypoint.…"   16 minutes ago   Up 15 minutes   0.0.0.0:8443->8443/tcp, 0.0.0.0:8081->80/tcp   web
f784156d6adb   postgres:14-alpine   "docker-entrypoint.s…"   16 minutes ago   Up 15 minutes   0.0.0.0:5433->5432/tcp
```
But when I try to access http://localhost:8443 or http://localhost:8081 in the browser of the host system I don't get any response. I didn't set the GOODLE_OAUTH2_CLIENT_ID and GOODLE_OAUTH2_CLIENT_SECRET, could this be the reason why?

> However, you can have a look at the flow on our staging environment
Which network do I have to use for your staging environment?

**Comment by samelamin** [comment]
> > Are you using an M1 Mac by any chance?
> 
> Yes, indeed. I confirm that I was able to start the containers on a legacy (non-M1) macbook instead:
> 
> ```shell
> $ docker ps
> CONTAINER ID   IMAGE                COMMAND                  CREATED          STATUS          PORTS                                          NAMES
> 851efd178a22   dapp_web             "/docker-entrypoint.…"   16 minutes ago   Up 15 minutes   0.0.0.0:8443->8443/tcp, 0.0.0.0:8081->80/tcp   web
> f784156d6adb   postgres:14-alpine   "docker-entrypoint.s…"   16 minutes ago   Up 15 minutes   0.0.0.0:5433->5432/tcp
> ```
> 
> But when I try to access http://localhost:8443 or http://localhost:8081 in the browser of the host system I don't get any response. I didn't set the GOODLE_OAUTH2_CLIENT_ID and GOODLE_OAUTH2_CLIENT_SECRET, could this be the reason why?
> 
> > However, you can have a look at the flow on our staging environment
> 
> Which network do I have to use for your staging environment?

Hi @takahser 

Thats really strange, the docker-compose file has multiple services that don't seem to be up. Maybe trying the [staging ](https://github.com/ImbueNetwork/dapp/blob/main/docker-compose.staging.yml#L1) compose file?


Also, the staging network link is `testnet.imbue.network`. That is what the staging URL is using.

So you can access the network via vanilla [PolkadotJs](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Ftestnet.imbue.network#/explorer) 

Perhaps it might make sense to validate the collator flow first, then focus on getting the UI done? Because the UI needs Google creds to work end to end. That's why I suggested you use the staging URL, it will add a project for you onto the staging network

If you create an account on the network, I can send you some $IMBU to enable you to create a project

**Comment by samelamin** [comment]
@takahser Before we got super excited, does that mean our Milestone 1 was accepted 😅 
If not, what is required to be done?

**Comment by takahser** [comment]
Thanks for your patience!

> If you create an account on the network, I can send you some $IMBU to enable you to create a project

Thanks, I was able to transfer them myself. I was also able to test the whole proposal process on your staging network.

> Maybe remove everything, clean checkout.

That actually did the trick. I also had the same issue with the collator docker setup. There, the launch container didn't start at first. After `docker-compose down`ing the whole thing and running `docker-compose up` again it worked. However, for some reason the _Extrinsics_  link was missing in the _Developer_ menu, I'm not sure why.

I also made some PRs containing some small improvements in your docs:
- https://github.com/ImbueNetwork/imbue/pull/15
- https://github.com/ImbueNetwork/dapp/pull/18

Finally, I finished my evaluation which you can find here:
- https://github.com/w3f/Grant-Milestone-Delivery/pull/356

In summary I'm very satisfied with your delivery. Feel free to improve on the things I commented on.

> does that mean our Milestone 1 was accepted 😅

As for the acceptance of M1, pls allow for some time so my team can look into it. As soon as it's merged it's officially accepted. :)

**Comment by samelamin** [comment]
> @samelamin thanks for your patience, I just finished my evaluation!
> 
> > If you create an account on the network, I can send you some $IMBU to enable you to create a project
> 
> Thanks, I was able to transfer them myself. I was also able to test the whole proposal process on your staging network.
> 
> > Maybe remove everything, clean checkout.
> 
> That actually did the trick. I also had the same issue with the collator docker setup. There, the launch container didn't start at first. After `docker-compose down`ing the whole thing and running `docker-compose up` again it worked. However, for some reason the _Extrinsics_ link was missing in the _Developer_ menu, I'm not sure why.
> 
> I also made some PRs containing some small improvements in your docs:
> 
> * [fix some typos and general improvements ImbueNetwork/imbue#15](https://github.com/ImbueNetwork/imbue/pull/15)
> * [fix typo ImbueNetwork/dapp#18](https://github.com/ImbueNetwork/dapp/pull/18)
> 
> Finally, I finished my evaluation which you can find here:
> 
> * [imbue M1 eval #356](https://github.com/w3f/Grant-Milestone-Delivery/pull/356)
> 
> In summary I'm very satisfied with your delivery. Feel free to improve on the things I commented on.
> 
> > does that mean our Milestone 1 was accepted 😅
> 
> As for the acceptance of M1, pls allow for some time so my team can look into it. As soon as it's merged it's officially accepted. :)

This is amazing @takahser, thank you so much for your feedback and your PRs. They are much appreciated! 
 >> As for the acceptance of M1, pls allow for some time so my team can look into it. As soon as it's merged it's officially accepted. :)

Of course, you can have all the time you need, but just to clarify, when you say "look into it" what do you mean by "it", perhaps review your evaluation? Or does someone else need to do the same evaluation, i.e multiple evaluations like we had for the grant proposal?

**Comment by takahser** [comment]
> Of course, you can have all the time you need, but just to clarify, when you say "look into it" what do you mean by "it", perhaps review your evaluation? Or does someone else need to do the same evaluation, i.e multiple evaluations like we had for the grant proposal?

Exactly, they'll have a look at my evaluation and if there are no objections it will be merged. There will be no additional evaluation. Should be fairly quick. 👍

**Comment by samelamin** [comment]
> > Of course, you can have all the time you need, but just to clarify, when you say "look into it" what do you mean by "it", perhaps review your evaluation? Or does someone else need to do the same evaluation, i.e multiple evaluations like we had for the grant proposal?
> 
> Exactly, they'll have a look at my evaluation and if there are no objections it will be merged. There will be no additional evaluation. Should be fairly quick. 👍

Thank you again @takahser! You are a star

**Comment by takahser** [comment]
@samelamin by the way, one thing I forgot to mention and could be improved: In your delivery table, the _Link_ and _Notes_ columns are mixed up starting from the 2nd entry _0a License_.

**Comment by samelamin** [comment]
> @samelamin by the way, one thing I forgot to mention and could be improved: In your delivery table, the _Link_ and _Notes_ columns are mixed up starting from the 2nd entry _0a License_.

Thanks @takahser  this has now been resolved

**Comment by samelamin** [comment]
> hi @samelamin we transferred the payment today. Thanks!

Received. Thank you so much!
